homeassistant/components/sensibo/diagnostics.py

```diff
@@ -3,16 +3,35 @@
 
 from typing import Any
 
+from homeassistant.components.diagnostics.util import async_redact_data
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 
 from .const import DOMAIN
 from .coordinator import SensiboDataUpdateCoordinator
 
+TO_REDACT = {
+    "location",
+    "ssid",
+    "id",
+    "macAddress",
+    "parentDeviceUid",
+    "qrId",
+    "serial",
+    "uid",
+    "email",
+    "firstName",
+    "lastName",
+    "username",
+    "podUid",
+    "deviceUid",
+}
+
 
 async def async_get_config_entry_diagnostics(
     hass: HomeAssistant, entry: ConfigEntry
 ) -> dict[str, Any]:
     """Return diagnostics for a config entry."""
     coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
-    return coordinator.data
+    get_all_data = await coordinator.client.async_get_devices()
```

Comment:
It would be nice to not refetch the data here. You could change the coordinator to instead return a dataclass with the `raw` and `processed` data.

Reply:
On the processed data I know what's there but on the raw I don't know hence why I wanted to get all if there is issues.
I can of course put the raw data also in the coordinator data so it's easily available but making a dataclass would be difficult as I don't know all which is in.

Reply:
I was thinking something simple that could be expanded later like:

```python3
from dataclasses import dataclass

@dataclass
class SensiboData:
   raw: dict
   parsed: dict

```